OpenRA.Mods.Common/Widgets/Logic/Editor/ActorSelectorLogic.cs

```diff
@@ -94,6 +86,24 @@ public ActorSelectorLogic(Widget widget, World world, WorldRenderer worldRendere
 				ownersDropDown.ShowDropDown("LABEL_DROPDOWN_TEMPLATE", 270, owners, setupItem);
 			};
 
+			editorLayer.OnSpawnRemoved = () =>
+			{
+				if (editorLayer.Players.Players.Values.Any(p => p.Name.Equals(selectedOwner.Name)))
+				{
+					return;
+				}
+
+				var orderedOwners = editorLayer.Players.Players.Values.ToList();
+				orderedOwners.Add(selectedOwner);
+				orderedOwners = orderedOwners.OrderBy(p => p.Name).ToList();
+
+				var indexofPreviousSelectedOwned = orderedOwners.IndexOf(selectedOwner);
+
+				SelectOwner(indexofPreviousSelectedOwned > 0
+					? orderedOwners.ElementAt(indexofPreviousSelectedOwned - 1)
+					: orderedOwners.FirstOrDefault());
```

Comment:
This is bad since orderedOwners.FirstOrDefault can still return a null reference if orderedOwners is empty, and SelectOwner assumes that it will get a valid option so it crashes on null.

Reply:
You could probably fix it by doing a null check in SelectOwner and then setting everything to default values if it isnt valid.
Unless orderedOwners can never be empty in which case ignore my comment.

Reply:
Actually I had a look and we assume above that orderedOwners can never be empty so change it from FirstOrDefault to just First and we will call it a day.